### PR TITLE
Make repo role bindings aware of projects

### PR DESCRIPTION
**Pull request by albscui**

### CODE REVIEW

File: src/server/pfs/server/driver.go [hunk 1]
```diff
@@ -219,7 +219,7 @@ func (d *driver) createRepo(txnCtx *txncontext.TransactionContext, repo *pfs.Rep
 				txnCtx,
 				whoAmI.Username,
 				[]string{auth.RepoOwnerRole},
-				&auth.Resource{Type: auth.ResourceType_REPO, Name: repo.Name},
+				&auth.Resource{Type: auth.ResourceType_REPO, Name: pfsdb.RepoKey(repo)},
```
Comment (robert-uhl) [inline src/server/pfs/server/driver.go:222]: Should this be the database key of the repo? That feels like using package `pfsdb` for something unrelated to the PFS database layer. Maybe this should be the result of `repo.String` instead?

File: src/server/pfs/server/testing/server_test.go [hunk 2]
```diff
@@ -310,6 +313,50 @@ func TestPFS(suite *testing.T) {
 		}
 	})
 
+	suite.Run("CreateRepoWithSameNameInDifferentProjects", func(t *testing.T) {
```
Comment (robert-uhl) [inline src/server/pfs/server/testing/server_test.go:316]: … AndAuth☺

**Comment by acohen4** [comment]
Reminder that this will need a migration as well

**Comment by acohen4** [comment]
I actually think it would make a lot of sense to do the Auth migration as a part of this PR @albscui

**Comment by albscui** [comment]
> I actually think it would make a lot of sense to do the Auth migration as a part of this PR @albscui

After some discussion, I think we agree that we will just do this in the big migrations work @robert-uhl is currently working on: https://github.com/pachyderm/pachyderm/pull/8245

**Review by robert-uhl — APPROVED** [review]

**Pull request merged**